java/yb-cql/src/test/java/org/yb/cql/CQLTester.java

```diff
@@ -0,0 +1,187 @@
+        return currentTable == null ? query : String.format(query, keyspace + "." + currentTable);
+    }
+
+    protected String keyspace()
+    {
+        return KEYSPACE;
+    }
+
+    protected String currentTable()
+    {
+        if (tables.isEmpty())
+            return null;
+        return tables.get(tables.size() - 1);
+    }
+
+    public ResultSet execute(String query, Object... values) throws Exception {
+        String fullQuery = formatQuery(KEYSPACE, query);
+        final ResultSet result = session.execute(fullQuery, values);
+        LOG.info("EXEC CQL FINISHED: {}", fullQuery);
+        return result;
+    }
+
+
+    public ResultSet execute(String query) throws Exception {
+        String fullQuery = formatQuery(KEYSPACE, query);
+        final ResultSet result = session.execute(fullQuery);
+        LOG.info("EXEC CQL FINISHED: {}", fullQuery);
+        return result;
+    }
+
+    protected void assertAllRows(Object[]... rows) throws Throwable
+    {
+        assertRows(execute("SELECT * FROM %s"), rows);
+    }
+
+    public static Object[] row(Object... expected)
+    {
+        return expected;
+    }
+
+    public static int displayRows(ResultSet result)
+    {
+        if (result == null)
+        {
+            LOG.info("Resultset has zero rows");
+            return 0;
+        }
+
+        ColumnDefinitions colDef = result.getColumnDefinitions();
+        List<ColumnDefinitions.Definition> meta = colDef.asList();
+        int i = 0;
+        for (Row actual : result)
+        {
+            LOG.info("Row {} = {}", i, actual.toString());
+            i++;
+        }
+	return i;
+    }
+
+    public static void assertRows(ResultSet result, Object[]... rows)
+    {
+        if (result == null)
+        {
+            if (rows.length > 0)
+                fail(String.format("No rows returned by query but %d expected", rows.length));
+            return;
+        }
+        ColumnDefinitions colDef = result.getColumnDefinitions();
+        List<ColumnDefinitions.Definition> meta = colDef.asList();
+        Iterator<Row> iter = result.iterator();
+        int i = 0;
+        if (!iter.hasNext() && rows.length > 0)
+	    fail(String.format("No rows returned by query but %d expected", rows.length));
+        while (iter.hasNext() && i < rows.length)
+        {
+            Object[] expected = rows[i];
+            Row actual = iter.next();
+
+            assertEquals(String.format("Invalid number of (expected) values provided for row %d", i), expected == null ? 1 : expected.length, meta.size());
+
+            for (int j = 0; j < meta.size(); j++)
+            {
+                ColumnDefinitions.Definition column = meta.get(j);
+                DataType.Name type = column.getType().getName();
+                switch (type) {
+                    case INT:
+                        assertEquals((Integer)expected[j], (Integer)actual.getInt(j));
+                        LOG.info("Row[{}, {}] = {}", i, j, actual.getInt(j));
```

Comment:
Please log before the `assertEquals`. Otherwise when assert fails, it aborts and we won't have this log message for the assert failure. Ditto for the same instance below.